Add HiVi Speaker integration documentation (#44021)
home-assistant/home-assistant.github.io: source/_integrations/hivi_speaker.markdown

---
title: HiVi Speaker
description: Instructions on how to integrate HiVi Multi-Room speakers into Home Assistant.
ha_category:
  - Media player
  - Switch

> ha_release: "2025.4"

Reviewer: ```suggestion
ha_release: 2025.4
```

ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:

>   - "@swansmart"

Author: Modified as suggested

ha_domain: hivi_speaker
ha_platforms:
  - media_player
  - switch

> ha_integration_type: hub

Author: Thanks for the review. This integration does discovery in its own code (custom network scan) and does not use Home Assistant's SSDP discovery framework, so I did not add `ha_ssdp: true` to the front matter. I have updated the document to remove the SSDP-related wording from the body and troubleshooting section so the description matches the integration behavior. The latest commit has been pushed.

---

> The **HiVi Speaker** {% term integration %} allows you to control [HiVi](https://www.hivi.com) Multi-Room speakers from Home Assistant. It discovers HiVi speakers on your local network and creates [media player](/integrations/media_player/) entities for playback control, plus [switch](/integrations/switch/) entities to manage multi-room synchronization (master–slave groups).

Reviewer: We avoid the master/slave terminology. Use alternatives such as server/client, leader/follower, or controller/device. Please fix throughout the page.
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/m/master-slave

Author: Thank you, I have made the suggested modifications.

The integration automatically discovers compatible HiVi Multi-Room devices on your local network and controls them directly over your home network. Media browsing can use local media and [DLNA Digital Media Server](/integrations/dlna_dms/) content when that integration is configured.

> 

Reviewer: ```suggestion

## Supported devices

Compatible with HiVi speakers that support Multi-Room, including (but not limited to):

- HiVi M5A, M3AMKIII, H6, H8, H5MKII, M500, M300MKII, M200MKII (Wi‑Fi), M200D, M100MKIII, M80W
- MT1-MAX, MT1-MINI
- T200MKII, MS2 series

Other HiVi models with Multi-Room functionality may also work.

```
We typically have this section after the introduction. https://developers.home-assistant.io/docs/documenting/integration-docs-examples/#basic-structure-of-an-integration-page

> This integration uses a custom SSDP discovery step (`ssdp:wiimudevice`) to find only HiVi Multi-Room–capable devices, then uses the manufacturer’s private HTTP protocol for sync control. Media browsing can use local media and [DLNA Digital Media Server](/integrations/dlna_dms/) content when that integration is configured.

Author: Modified as suggested

There is support for the following platform types within Home Assistant:

- [Media player](#media-player)

> - [Switch](#switch)

Reviewer: ```suggestion
```
I don't think this is needed here.

## Prerequisites

- HiVi speakers must be powered on and connected to the same local network as Home Assistant.
- Speakers must support Multi-Room functionality (see [Supported devices](#supported-devices)).

{% include integrations/config_flow.md %}

Setup is single-instance: you add the integration once. The integration then scans the network for HiVi Multi-Room speakers and creates entities as devices are discovered.

> ### Integration options

Author: Modified as suggested

> From the integration’s **Options** you can trigger a **Refresh discovery** to scan again for speakers. Newly discovered devices will be added automatically.

Reviewer: ```suggestion
From the integration’s **Options**, you can trigger a **Refresh discovery** to scan again for speakers. Newly discovered devices will be added automatically.
```

> ## Media player

Reviewer: ```suggestion
### Supported functionality

The **HiVi Speakers** integration provides the following entities.

### Media player
```
We usually have a supported functionality section here: https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L88

Each discovered HiVi speaker is represented by a [media player](/integrations/media_player/) entity. You can:

- **Play / Pause** — Control playback on that speaker.
- **Volume** — Adjust volume and mute.
- **Media source** — Choose media from local sources or from a configured [DLNA DMS](/integrations/dlna_dms/) (when available).

> - **Browse media** — Use the media browser to select and play music or other audio.

Reviewer: ```suggestion
- **Play / Pause**: Control playback on that speaker.
- **Volume**: Adjust volume and mute.
- **Media source**: Choose media from local sources or from a configured [DLNA DMS](/integrations/dlna_dms/) (when available).
- **Browse media**: Use the media browser to select and play music or other audio.
```

Playback and volume apply to the individual speaker. To play the same audio on multiple speakers in sync, use the [Multi-room synchronization](#multi-room-synchronization) switches below.

> ## Switch

Reviewer: ```suggestion
### Switches
```

For each HiVi speaker that can act as a **leader**, the integration creates **switch** entities: one per other available speaker. Each switch represents “use this other speaker as a synchronized **follower** of the current leader.”

- **Switch on** — Add that speaker as a follower; it will play in sync with the leader.

> - **Switch off** — Remove the follower link; that speaker returns to standalone mode.

Reviewer: ```suggestion
- **Switch on**: Add that speaker as a follower; it will play in sync with the leader.
- **Switch off**: Remove the follower link; that speaker returns to standalone mode.
```

> - **Switch OFF** — Remove the slave link; that speaker returns to standalone mode.

Author: Modified as suggested

You can change sync groups at any time; multiple independent master–slave groups are supported.

## Multi-room synchronization

Multi-room works by designating one speaker as the **master** and others as **slaves**. Audio played on the master is streamed in sync to all its slaves.

### How to set up

1. Open the HiVi speaker **device** in Home Assistant (the one you want to use as master).
2. In the device’s configuration area, find the list of switches named after other HiVi speakers.
3. Turn **ON** the switch for each speaker you want to sync with this master.
4. Those speakers become slaves of this master; playback on the master is synchronized to them.

> 5. Turn **OFF** a switch to remove that speaker from the sync group.

Author: Modified as suggested

### Example layout

> ```

Author: Modified as suggested

Living Room Speaker (master)
├── [ON]  Kitchen Speaker   (synchronized)
├── [ON]  Bedroom Speaker   (synchronized)
├── [OFF] Bathroom Speaker  (standalone)
└── [OFF] Office Speaker   (standalone)
```

- Each master can have its own set of slaves.
- All speakers in a sync group must be on the same local network.
- You can add or remove slaves at any time without restarting playback.

## Supported devices

Compatible with HiVi speakers that support Multi-Room, including (but not limited to):

> - HiVi M5A, M3AMKIII, H6, H8, H5MKII, M500, M300MKII, M200MKII (WiFi), M200D, M100MKIII, M80W

Author: Modified as suggested

- MT1-MAX, MT1-MINI
- T200MKII, MS2 series

Other HiVi models with Multi-Room functionality may also work.

> 

Reviewer: ```suggestion
```
I would move this section further up.

## Troubleshooting

### No speakers are found

- Confirm HiVi speakers are on and on the same network as Home Assistant.
- Ensure multicast is not blocked (UDP 239.255.255.250, port 1900).
- Restart Home Assistant and the speakers, then use **Options** > **Refresh discovery**.

### Speakers are out of sync during playback

- Update speaker firmware if possible.
- Improve network stability and reduce latency; fewer speakers in a sync group can help.
- Place speakers and router so that Wi‑Fi signal is strong.

### Speakers disconnect often

- Check Wi‑Fi signal strength.
- Consider assigning static IPs to speakers and updating router firmware.

### Enabling debug logging

To help diagnose issues, enable debug logging for this integration:

```yaml
# configuration.yaml
logger:
  default: warning
  logs:
    homeassistant.components.hivi_speaker: debug
```

Restart Home Assistant and reproduce the issue, then inspect the logs.

## Removing the integration

> Removing the integration from {% my integrations title="**Settings** > **Devices & services**" %} will remove all related entities and devices. No extra steps are required.

Reviewer: ```suggestion
This integration follows standard integration removal. No extra steps are required.

{% include integrations/remove_device_service.md %}
```

{% include integrations/config_flow.md %}

Setup is single-instance: you add the integration once. The integration then scans the network for HiVi Multi-Room speakers and creates entities as devices are discovered.

## Options

From the integration’s **Options**, you can trigger a **Refresh discovery** to scan again for speakers. Newly discovered devices will be added automatically.
### Supported functionality

The **HiVi Speakers** integration provides the following entities.

### Media player

Each discovered HiVi speaker is represented by a [media player](/integrations/media_player/) entity. You can:

- **Play / Pause**: Control playback on that speaker.
- **Volume**: Adjust volume and mute.
- **Media source**: Choose media from local sources or from a configured [DLNA DMS](/integrations/dlna_dms/) (when available).
- **Browse media**: Use the media browser to select and play music or other audio.

Playback and volume apply to the individual speaker. To play the same audio on multiple speakers in sync, use the [Multi-room synchronization](#multi-room-synchronization) switches below.

### Switches

> For each HiVi speaker that can act as a **leader**, the integration creates **switch** entities: one per other available speaker. Each switch represents “use this other speaker as a synchronized **follower** of the current leader.”

Reviewer: ```suggestion
For each HiVi speaker that can act as a **leader**, the integration creates **switch** entities: one per other available speaker. Each switch represents “use this other speaker as a synchronized **follower** of the current leader”.
```

> For each HiVi speaker that can act as a **leader**, the integration creates **switch** entities: one per other available speaker. Each switch represents “use this other speaker as a synchronized **follower** of the current leader”.

Reviewer: ```suggestion
For each HiVi speaker that can act as a **leader**, the integration creates **switch** entities: one per other available speaker. Each switch represents "use this other speaker as a synchronized **follower** of the current leader".
```

- **Switch on**: Add that speaker as a follower; it will play in sync with the leader.
- **Switch off**: Remove the follower link; that speaker returns to standalone mode.

You can change sync groups at any time; multiple independent leader–follower groups are supported.

## Multi-room synchronization

Multi-room works by designating one speaker as the **leader** and others as **followers**. Audio played on the leader is streamed in sync to all its followers.

### How to set up

> 1. Open the HiVi speaker **device** in Home Assistant (the one you want to use as leader).

Reviewer: ```suggestion
1. In Home Assistant, go to {% my integrations title="**Settings** > **Devices & services**" %}, and select the **HiVi Speaker** integration. From the list of devices, select the HiVi speaker **device** in Home Assistant (the one you want to use as leader).
```
Is is worth noting how to open the integration page? Please correct the steps if I got this wrong.

Author: You're right, these are the specific steps

2. In the device’s configuration area, find the list of switches named after other HiVi speakers.
3. Turn on the switch for each speaker you want to sync with this leader.
4. Those speakers become followers of this leader; playback on the leader is synchronized to them.
5. Turn off a switch to remove that speaker from the sync group.

### Example layout

```text
Living Room Speaker (leader)
├── [On]  Kitchen Speaker   (synchronized)
├── [On]  Bedroom Speaker   (synchronized)
├── [Off] Bathroom Speaker  (standalone)
└── [Off] Office Speaker   (standalone)
```

- Each leader can have its own set of followers.
- All speakers in a sync group must be on the same local network.
- You can add or remove followers at any time without restarting playback.

## Troubleshooting

### No speakers are found

- Confirm HiVi speakers are on and on the same network as Home Assistant.

> - Ensure multicast is not blocked (UDP 239.255.255.250, port 1900).

Reviewer: ```suggestion
- Ensure multicast is not blocked (UDP `239.255.255.250`, port `1900`).
```

- Restart Home Assistant and the speakers, then use **Options** > **Refresh discovery**.

### Speakers are out of sync during playback

- Update speaker firmware if possible.
- Improve network stability and reduce latency; fewer speakers in a sync group can help.
- Place speakers and router so that Wi‑Fi signal is strong.

### Speakers disconnect often

- Check Wi‑Fi signal strength.
- Consider assigning static IPs to speakers and updating router firmware.

### Enabling debug logging

> To help diagnose issues, enable debug logging for this integration:

Reviewer: Maybe add a link to the general debug instructions: https://www.home-assistant.io/docs/configuration/troubleshooting/#debug-logs-and-diagnostics
